kroxylicious-kms-provider-aws-kms/pom.xml

```diff
@@ -96,4 +96,31 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>io.kroxylicious</groupId>
+                <artifactId>kroxylicious-schema-maven-plugin</artifactId>
+                <version>${project.version}</version>
+                <executions>
+                    <execution>
+                        <id>generate-config-model</id>
+                        <goals>
+                            <goal>compile-schema</goal>
+                        </goals>
+                        <phase>generate-sources</phase>
+                        <configuration>
+                            <failOnWarnings>true</failOnWarnings>
+                            <headerFile>../etc/license.txt</headerFile>
+                            <existingClasses>
```

Comment:
This is a bit of a hack which allows use to continue using the existing Java types for these things, while still exposing the right JsonSchema. But there's some tech debt being added here because both this and the Vault filter duplicate those Json Schemas. The way to fix that is have some common json for common types, which we include here via a `$ref`.

Reply:
agreed, but we can probably address the tech debt as a separate PR.